src/native/managed/cdacreader/src/Legacy/SOSDacImpl.cs

```diff
@@ -440,10 +440,13 @@ public unsafe int GetObjectData(ulong objAddr, DacpObjectData* data)
                 // Update the size to include the array components
                 data->Size += numComponents * data->dwComponentSize;
 
-                // Get the type of the array elements
+                // Get the type handle of the array elements
                 TypeHandle element = runtimeTypeSystemContract.GetTypeParam(handle);
                 data->ElementTypeHandle = element.Address;
-                data->ElementType = (uint)runtimeTypeSystemContract.GetSignatureCorElementType(element);
+
+                // Get the element type from the canonical method table - see MethodTable::GetArrayElementType for coreclr equivalent
```

Comment:
This is historic artifact of how arrays were represented in the past and fragile NGen.

It may be best to delete `ArrayClass::m_ElementType` field, `GetArrayElementType` methods, and replace their calls with `GetArrayElementTypeHandle().GetSignatureCorElementType/GetVerifierCorElementType` as appropriate.